testsuite/scripts/src/test/java/org/wildfly/launcher/test/LauncherTestCase.java

```diff
@@ -0,0 +1,223 @@
+ * limitations under the License.
+ */
+
+package org.wildfly.launcher.test;
+
+import static org.jboss.as.controller.client.helpers.ClientConstants.CONTROLLER_PROCESS_STATE_STARTING;
+import static org.jboss.as.controller.client.helpers.ClientConstants.CONTROLLER_PROCESS_STATE_STOPPING;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.StringReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import javax.json.Json;
+import javax.json.JsonObject;
+import javax.json.JsonReader;
+
+import org.jboss.as.controller.client.ModelControllerClient;
+import org.jboss.as.controller.client.Operation;
+import org.jboss.as.controller.client.OperationMessageHandler;
+import org.jboss.as.controller.client.OperationResponse;
+import org.jboss.as.controller.client.helpers.Operations;
+import org.jboss.as.test.shared.TestSuiteEnvironment;
+import org.jboss.as.test.shared.TimeoutUtil;
+import org.jboss.dmr.ModelNode;
+import org.junit.Assert;
+import org.junit.Test;
+import org.wildfly.core.launcher.Launcher;
+import org.wildfly.core.launcher.ProcessHelper;
+import org.wildfly.core.launcher.StandaloneCommandBuilder;
+
+/**
+ * This tests launching WildFly with JBoss Modules as an agent. For details see
+ * <a href="https://issues.jboss.org/browse/WFCORE-4674">WFCORE-4677</a>.
+ *
+ * @author <a href="mailto:jperkins@redhat.com">James R. Perkins</a>
+ */
+@SuppressWarnings("MagicNumber")
+public class LauncherTestCase {
+    private static final ModelNode EMPTY_ADDRESS = new ModelNode().setEmptyList();
+    private static final Path STDOUT_PATH = Paths.get(TestSuiteEnvironment.getTmpDir(), "stdout.txt");
+    private static final String JBOSS_HOME = System.getProperty("jboss.home");
+    private static final int DEFAULT_TIMEOUT = TimeoutUtil.adjust(Integer.parseInt(System.getProperty("jboss.test.start.timeout", "15")));
+
+    static {
+        EMPTY_ADDRESS.protect();
+    }
+
+    @Test
+    public void testStandaloneWithAgent() throws Exception {
+        final StandaloneCommandBuilder builder = StandaloneCommandBuilder.of(JBOSS_HOME)
+                .setUseSecurityManager(parseProperty("security.manager"))
+                // Add the test logging agent to the jboss-modules arguments
+                .addAgent(Paths.get(JBOSS_HOME, "logging-agent-tests.jar").toString());
+
+        Process process = null;
+        try {
+            process = Launcher.of(builder)
+                    .setRedirectErrorStream(true)
+                    .redirectOutput(STDOUT_PATH)
+                    .launch();
+            waitForStandalone(process);
+            final List<JsonObject> lines = readLogFileFromModel("json.log");
+            Assert.assertEquals("Expected 1 line found " + lines.size(), 1, lines.size());
```

Comment:
This would fail if this test is not run first. We should see if set `append` to `false` and get this working in the CLI commands.